ratik/src/intTest/java/org/opencadc/ratik/InventoryValidatorTest.java

```diff
@@ -0,0 +1,560 @@
+import org.opencadc.inventory.DeletedStorageLocationEvent;
+import org.opencadc.inventory.SiteLocation;
+import org.opencadc.inventory.StorageSite;
+
+public class InventoryValidatorTest {
+    private static final Logger log = Logger.getLogger(InventoryValidatorTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.ratik", Level.INFO);
+        Log4jInit.setLevel("org.opencadc.inventory", Level.INFO);
+        Log4jInit.setLevel("org.opencadc.inventory.db", Level.INFO);
+        Log4jInit.setLevel("ca.nrc.cadc.db", Level.INFO);
+    }
+
+    private final InventoryEnvironment localEnvironment = new InventoryEnvironment();
+    private final LuskanEnvironment remoteEnvironment = new LuskanEnvironment();
+
+    public InventoryValidatorTest() throws Exception {
+
+    }
+
+    @Before
+    public void beforeTest() throws Exception {
+        this.localEnvironment.cleanTestEnvironment();
+        this.remoteEnvironment.cleanTestEnvironment();
+    }
+
+    /**
+     * discrepancy: none
+     * before: Artifact in L & R
+     * after: Artifact in L & R
+     */
+    @Test
+    public void noDiscrepancy() throws Exception {
+        Calendar calendar = Calendar.getInstance();
+        Artifact artifact = new Artifact(URI.create("cadc:INTTEST/one.ext"), TestUtil.getRandomMD5(), calendar.getTime(), 1024L);
+        this.localEnvironment.artifactDAO.put(artifact);
+        this.remoteEnvironment.artifactDAO.put(artifact);
+
+        InventoryValidator testSubject = new InventoryValidator(this.localEnvironment.daoConfig, TestUtil.LUSKAN_URI, artifact.getBucket());
+        testSubject.run();
+
+        Artifact localArtifact = this.localEnvironment.artifactDAO.get(artifact.getURI());
+        Assert.assertNotNull("local artifact not found", localArtifact);
+        Assert.assertTrue("local artifact changed", localArtifact.equals(artifact));
+        // check other artifact attributes?
+    }
+
+    // discrepancy: artifact in L && artifact not in R //
+
+    /** discrepancy: artifact in L && artifact not in R
+     * explanation0: filter policy at L changed to exclude artifact in R
+     * evidence: Artifact in R without filter
+     * action: delete Artifact, if (L==storage) create DeletedStorageLocationEvent
+     *
+     * changed filter policy, global doesn't filter, means L == storage only?
+     * before: Artifact in L & R, filter policy to exclude Artifact in R
+     * after: Artifact not in L, DeletedStorageLocationEvent in L
+     */
+    @Test
+    public void explanation0_inL() throws Exception {
+        Calendar calendar = Calendar.getInstance();
+        Artifact artifact = new Artifact(URI.create("cadc:INTTEST/one.ext"), TestUtil.getRandomMD5(), calendar.getTime(), 1024L);
+        this.localEnvironment.artifactDAO.put(artifact);
+        this.remoteEnvironment.artifactDAO.put(artifact);
+
+        //.necessary to carefully generate a uri bucket, or will an out of range bucket work?
```

Comment:
filter policy excludes artifact means a config/artifact-filter.sql (like fenwick) containing a WHERE clause that does not match the remote artifact.

So, you do need an artifact-filter.sql file in {user.home}/config... should be in build/tmp like ringhold tests

Reply:
Missed responding to these last comments. My uri bucket knowledge is still stuck on the AD site where the uri bucket IS the archive. When local is a storage site, the filter policy will include the storage site archives from global, and uri bucket could be empty to include all artifacts on the storage site. When local is global, the filter policy could be empty to include all storage site artifacts, and the uri bucket would somehow restrict the global artifacts to only those on the storage site. That last bit is what I'm not fully understanding yet. Need to look at global and a ceph site to see how that works.

Added support for the artifact-filter.sql config file.